oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/parser/listener/strategy/ExceptionAnalyzer.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.analyzer.provider.trace.parser.listener.strategy;
+
+import org.apache.skywalking.apm.network.language.agent.v3.SegmentObject;
+
+/**
+ * judge the status of spanObject
+ */
+public interface ExceptionAnalyzer {
+    /**
+     * extract error status from the span collection in segmentObject
+     */
+    boolean isError(SegmentObject segmentObject);
```

Comment:
This is a low-performance method design. In the `SegmentAnalysisListener#parseSegment`, we have the span list iteration already, this interface forces the implementation to do it again. 
```suggestion
    boolean isError(SpanObject span);
```